Sources/App/Models/Package.swift

```diff
@@ -215,6 +234,35 @@ extension Package {
 }
 
 
+extension Package {
+    func updatePlatformCompatibility(on database: Database) -> EventLoopFuture<Void> {
+        guard let db = database as? SQLDatabase else {
+            return database.eventLoop.future(error: AppError.genericError(id, "Database must be an SQLDatabase ('as? SQLDatabase' must succeed)"))
+        }
+        return db.raw(
+            #"""
+            UPDATE packages p SET platform_compatibility = ARRAY(
+                SELECT
+                    CASE
+                        WHEN b.platform LIKE 'macos-%' THEN 'macos'
+                        ELSE b.platform
+                    END
+                FROM versions v
+                JOIN builds b ON b.version_id = v.id
+                WHERE v.package_id = p.id
+                AND v.latest IS NOT NULL
+                AND b.status = 'ok'
+                GROUP BY b.platform
+                HAVING count(*) > 0
+            ),
+            updated_at = NOW()
+            WHERE p.id = \#(bind: id)
```

Comment:
I've tried to write this update statement in Fluent but if it's possible at all it's going to be hard and potentially require upstream changes. But more importantly I believe there'd be some uncertainty whether that update runs on the server side or if it'd pulls records and send another update statement back.

Using raw SQL we _know_ we're doing it in the most efficient way. The downside is losing type-safety (see below).